### DISCUSSION TITLE
Seed database with relational fields

### DISCUSSION THREAD

**Discussion by kurt-hardy** [prisma/prisma#2247] [Q&A] [answered]
Hi

Im trying to seed a postgres database with country and region. Creating a country is find but I'm getting errors creating the region. Here is my create:

```ts
const on = await prisma.region.create({
  data: { name: 'on', country: { connect: canada } }
});
```
I keep getting this error: Invalid `prisma.region.create()` invocation:

This schema is straight forward, country is id and name, region is id, name and countryId

Heres the full error:

```js
{
  data: {
    name: 'on',
    country: {
      connect: {
        id: 'test id',
        createdAt: new Date('2020-04-20T21:23:57.736Z'),
        updatedAt: new Date('2020-04-20T21:23:57.736Z'),
        name: 'UK'
        ~~~~
      }
    }
  }
}
```

Unknown arg `createdAt` in data.country.connect.createdAt for type CountryWhereUniqueInput. Available args:

```ts
type CountryWhereUniqueInput {
  id?: String
}
Unknown arg `updatedAt` in data.country.connect.updatedAt for type CountryWhereUniqueInput. Available args:

type CountryWhereUniqueInput {
  id?: String
}
Unknown arg `name` in data.country.connect.name for type CountryWhereUniqueInput. Did you mean `id`? Available args:
type CountryWhereUniqueInput {
  id?: String
}
```

Thanks for your help

**Comment by Rizialdi** [comment, marked as answer]
Hi, 
the error is basically saying that you're trying to connect to a field which is not found.
You cannot connect to canada, instead, you should connect to a reference to canada in your database.
Try to change:
```js
const on = await prisma.region.create({ data: { name: 'on', country: { connect: canada } } });
```
to 
```js
const on = await prisma.region.create({ data: { name: 'on', country: { connect: { id: 's4f5d45sd45sd' } } });
// with s4f5d45sd45sd is a Unique Identifier for a field in Country type
// namely *id* in your case 
```

```prisma
Unknown arg name in data.country.connect.name for type CountryWhereUniqueInput. Did you mean id? Available args:
type CountryWhereUniqueInput {
id?: String // <- the only available identifier
}
```

Also, for a straight forward seed file, checkout this [example](https://github.com/prisma/prisma-examples/blob/master/experimental/javascript/graphql/prisma/seed.js) in the docs. You can replace the field with your data (**user1** and **user 2** to your **regions** and fields inside **posts** to your **countries** )

Hope this helps.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
